From 4aa2c7f4ae4e7c4d7712c90e982642d7b4d59911 Mon Sep 17 00:00:00 2001
From: Adam Ratcliffe <adam.ratcliffe@tomtom.com>
Date: Fri, 8 Nov 2024 11:48:47 -0800
Subject: [PATCH 07/22] Handle CSS and JSON imports.

---
 package.json      |  2 ++
 rollup.config.mjs | 11 ++++++++---
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/package.json b/package.json
index 6f57ae2..42cd2ea 100755
--- a/package.json
+++ b/package.json
@@ -48,8 +48,10 @@
     "@rollup/plugin-commonjs": "^28.0.1",
     "@rollup/plugin-json": "^6.1.0",
     "@rollup/plugin-node-resolve": "^15.3.0",
+    "postcss": "^8.4.47",
     "rollup": "^4.24.4",
     "rollup-plugin-peer-deps-external": "^2.2.4",
+    "rollup-plugin-postcss": "^4.0.2",
     "rollup-plugin-terser": "^7.0.2"
   },
   "peerDependencies": {
diff --git a/rollup.config.mjs b/rollup.config.mjs
index 89ae5dd..4700c8c 100644
--- a/rollup.config.mjs
+++ b/rollup.config.mjs
@@ -1,6 +1,7 @@
 import babel from "@rollup/plugin-babel";
 import resolve from "@rollup/plugin-node-resolve";
 import commonjs from "@rollup/plugin-commonjs";
+import postcss from "rollup-plugin-postcss";
 import json from "@rollup/plugin-json";
 import peerDepsExternal from "rollup-plugin-peer-deps-external";
 
@@ -20,9 +21,9 @@ export default {
   ],
   plugins: [
     peerDepsExternal(),
-    // resolve({
-    //   extensions: [".js", ".jsx"]
-    // }),
+    resolve({
+      extensions: [".js", ".jsx"]
+    }),
     babel({
       exclude: "node_modules/**",
       presets: ["@babel/preset-react"],
@@ -30,6 +31,10 @@ export default {
       extensions: [".js", ".jsx"]
     }),
     commonjs(),
+    postcss({
+      extract: true, // This will extract the CSS to a separate file
+      minimize: true // Optionally minimize the CSS
+    }),
     json()
   ]
 };
-- 
2.39.5 (Apple Git-154)

