From 5a8eda122cf48a226414fc9b3f4315cdeb4436cf Mon Sep 17 00:00:00 2001
From: Adam Ratcliffe <adam.ratcliffe@tomtom.com>
Date: Fri, 8 Nov 2024 14:34:54 -0800
Subject: [PATCH 12/22] For modules specific to Node.js, exclude from rollup
 config.

---
 rollup.config.mjs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/rollup.config.mjs b/rollup.config.mjs
index 44f49cc..e63bc07 100644
--- a/rollup.config.mjs
+++ b/rollup.config.mjs
@@ -37,5 +37,6 @@ export default {
       minimize: true // Optionally minimize the CSS
     }),
     json()
-  ]
+  ],
+  external: ["worker_threads", "react-dom/server"]
 };
-- 
2.39.5 (Apple Git-154)

