From 8d9dd869293433f1c3600845a658ce94f8af6377 Mon Sep 17 00:00:00 2001
From: Adam Ratcliffe <adam.ratcliffe@tomtom.com>
Date: Fri, 8 Nov 2024 14:45:02 -0800
Subject: [PATCH 14/22] Move react-dom/server also.

---
 package.json      | 3 ++-
 rollup.config.mjs | 3 +--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/package.json b/package.json
index bea5d51..2784b50 100755
--- a/package.json
+++ b/package.json
@@ -67,6 +67,7 @@
   },
   "browser": {
     "crypto": false,
-    "worker_threads": false
+    "worker_threads": false,
+    "react-dom/server": false
   }
 }
\ No newline at end of file
diff --git a/rollup.config.mjs b/rollup.config.mjs
index 14fb668..44f49cc 100644
--- a/rollup.config.mjs
+++ b/rollup.config.mjs
@@ -37,6 +37,5 @@ export default {
       minimize: true // Optionally minimize the CSS
     }),
     json()
-  ],
-  external: ["react-dom/server"]
+  ]
 };
-- 
2.39.5 (Apple Git-154)

