From 8de511bed9ed0dc13d963569d46489dcfd4a5b38 Mon Sep 17 00:00:00 2001
From: Adam Ratcliffe <adam.ratcliffe@tomtom.com>
Date: Fri, 8 Nov 2024 15:25:47 -0800
Subject: [PATCH 19/22] Default to en-US for language.

---
 src/app/App.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/app/App.js b/src/app/App.js
index a52a0c7..6759243 100644
--- a/src/app/App.js
+++ b/src/app/App.js
@@ -30,7 +30,7 @@ import {
 function App({
   apiKey,
   theme,
-  language,
+  language = "en-US",
   measurementSystem,
   width,
   height,
-- 
2.39.5 (Apple Git-154)

